pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1130,8 +1130,8 @@ def _looks_like_infinite_iterator(param: nodes.NodeNG) -> bool:
                 return inferred.qname() in KNOWN_INFINITE_ITERATORS
             return False
 
-        if isinstance(node.func, nodes.Attribute):
-            # A next() method, which is now what we want.
+        if isinstance(node.func, nodes.Attribute) or not node.args:
```

Comment:
Shouldn't we use inference here to be sure that the next is the one we want ? If we have an inner ``next`` function with no args we'd still have buggy behavior, right ?

Reply:
Please see the PR description where I address this. Inference _is_ being used in this method, the problem is that its output is incorrect and I'm not sure how to fix it? If you have suggestions, with code, that may help!

Reply:
This is something that should preferably fixed in astroid then. If we merge this because the astroid fix is hard to do or impossible then we need to document that it's unnecessary if astroid inference get better.

Reply:
Turns out I was wrong. Inference is happening correctly. Even though an author writes a `next` function, astroid correctly infers it to the custom next, not the builtin next.

This solution turned out to be correct. we shouldn't assume `node.args` is populated. I also originally assumed this would NOT work for a custom next with args, but I added a test to demonstrate that it does.